cfme/tests/test_utilization.py

```diff
@@ -28,6 +29,10 @@ def handle_provider(provider_key):
     try:
         providers.clear_providers()
         providers.setup_provider(provider_key)
+    except FlashMessageException as e:
+        logger.error("Could not set up the provider due to: {}".format(str(e)))
+        pytest.skip("Skipping due to provider setup failure: {}".format(str(e)))
```

Comment:
Could we put this in a function somewhere? Maybe on `FlashMessageException` itself? I'd like to see the repetition of this except block go away. So maybe on the `FlashMessageException` class add something like a  `provider_setup_failed` method that does the logging, and raises skip if a `skip` kwarg is `True`?